### ISSUE TITLE
merge auth.User and users.User models

### ISSUE THREAD

**Issue by arush0311** [mozilla/api.mozillascience.org#79]
We use auth.User model for admin stuff and users.User for storing members details. There will be two other types of users added in the near future normal users which will use science website and blog writers. I propose that we merge all of them in one models so django auth package can take care of them by default.

**Comment by alanmoo** [comment]
@Pomax I believe you might have thoughts on this based on the work you were recently doing in the network api. Did you say something along the lines of it not making sense to join authentication and user data?

**Comment by Pomax** [comment]
Correct, you want to keep the user model fairly clean, with only data relevant to log in, so your User model only has stuff like username, email, password, etc, and then member details go in a UserProfile or similarly sensibly named model, which can then refer to a user with a 1-to-1 relation (using a ForeignKeyField). You can then automatically set up a UserProfile when a user doesn't have one using either a "pipeline fuction" (if the authentication process uses a pipeline, such as the social auth uses, example of that in the network-api [here](https://github.com/mozilla/network-api/blob/master/app/networkapi/settings.py#L160) and [here](https://github.com/mozilla/network-api/blob/master/app/networkapi/utility/userpermissions.py)) or a using a signal handler (example of that in the network-api [here](https://github.com/mozilla/network-api/blob/master/app/pulseapi/userprofile/__init__.py), [here](https://github.com/mozilla/network-api/blob/master/app/pulseapi/userprofile/apps.py), and [here](https://github.com/mozilla/network-api/blob/master/app/pulseapi/userprofile/signals.py) -- it requires three files rather than one file)

**Comment by arush0311** [comment]
ah. so we need to refer the auth.user from users.user models and remove some duplicate data. Add users.user entries to auth.user and thats it I think.

**Comment by Pomax** [comment]
don't put anything "in" auth.user -- Django creates a "soft" property on your user when you use foreign links. For example: https://github.com/mozilla/network-api/blob/master/app/pulseapi/userprofile/models.py#L16-L20 sets up a link between a user and a user profile. From then on, if you have a user, you can get their associated profile as:
```
profile = UserProfile.objects.get(user = someuserobject)
```
And if you have a profile you can get the associated user as
```
user = profile.user
```

**Comment by arush0311** [comment]
@pomax Actually users in auth.User and users.User are completely independent of each other, I was talking about adding entries to auth.user which are not present in profile.user.

**Comment by alanmoo** [comment]
I'm not sure if it was made clear previously, but the goal is to allow anyone to log into the site via github authentication. At that point, all Auth users will have a Users.user profile (even if they're not involved in any project), and auth users will be everyone that's logged in to claim themselves as working on a project.

Does that clear anything up? What are you looking to add to auth.user?

**Comment by arush0311** [comment]
But right now there are users,User entries which are not in auth.User right? I was saying to add those entries to auth.User and remove redundant data from it like username and such.

**Comment by alanmoo** [comment]
Auth.users should only be people who have logged in to the site. Right now that's only possible if we create an admin account for them; in the future it will happen automatically when they log in using Github OAuth. @auremoser has pointed out there is some redundant data in the Users.user model, but as @Pomax stated, we should keep them as related models.

cc @acabunoc

**Comment by arush0311** [comment]
And the redundant data be removed? Can you tell me what is stored right now in user.User model?
